cmake/std/atdm/utils/get_known_system_info.sh

```diff
@@ -11,127 +11,196 @@ unset ATDM_CONFIG_CDASH_HOSTNAME
-        # so select the system name 'sem-rhel6'
-        ATDM_SYSTEM_NAME=cee-rhel6
-        ATDM_HOSTNAME=cee-rhel6
-      else
-        echo
-        echo "***"
-        echo "*** Error, hostname='$ATDM_CONFIG_REAL_HOSTNAME' is a 'sems-rhel6' machine but is"
-        echo "*** is *not* a 'cee-rhel6' machine but 'cee-rhel6' was given in the"
-        echo "*** build name string '$ATDM_CONFIG_BUILD_NAME'!  Please remove 'cee-rhel6'"
-	echo "*** from the build name or provide 'sems-rhel6' and then the 'sems-rhel6'"
-	echo "*** env will be used."
-        echo "***"
-        return
-      fi
-    else
-      # 'cee-rhel6' nor 'sems-rhel6' was set in the build name, so the default
-      # is to sue the 'sems-rhel6' env!
-      ATDM_HOSTNAME=sems-rhel6
-      ATDM_SYSTEM_NAME=sems-rhel6
+  elif [[ "${ATDM_SEMS_SYSTEM_NAME}" == "rhel7-x86_64" ]] ; then
+    systemNameTypeMatchedList+=(sems-rhel7)
+    systemNameTypeMatchedListHostNames[sems-rhel7]=sems-rhel7
+    if [[ -d /projects/sparc/tpls ]] ; then
+      systemNameTypeMatchedList+=(cee-rhel6)
+      systemNameTypeMatchedListHostNames[cee-rhel6]=cee-rhel6
     fi
-  else
-    echo
-    echo "***"
-    echo "*** Error, hostname='$ATDM_CONFIG_REAL_HOSTNAME' has the SEMS env"
-    echo "*** mounted but the SEMS system '${ATDM_SYSTEM_NAME}' is not yet supported!"
-    echo "***"
-    return
+  # Else: We don't support any other SEMS systems yet!
   fi
 fi
 
+# If the user puts 'spack-rhel' in the build name, assume that the modules are
+# there.
+if [[ "${knownSystemNameInBuildName}" == "spack-rhel" ]] ; then
+  systemNameTypeMatchedList+=(spack-rhel)
+  systemNameTypeMatchedListHostNames[spack-rhel]=spack-rhel
+fi
+# NOTE: Above, I was using:
+#
+#  spackCMakeModule=`module avail 2>&1 | grep spack-cmake | head -1`
+#
+# to determine if a spack-rhel env was setup but that check actually takes a
+# long time (5 sec) on some systems so I went with the easier logic above.
+
+#echo "systemNameTypeMatchedList = '(${systemNameTypeMatchedList[@]})'"
+
+#################################################################################
+### NOTE: No modifications below this line should be needed when adding a new
+### system base on hostname or system type!
+#################################################################################
+
+#
+# D) Select a known system given the above info
+#
+
+ATDM_HOSTNAME=
+ATDM_SYSTEM_NAME=
+
+# D.1) First, go with the system name in the build name.
+if [[ "${ATDM_SYSTEM_NAME}" == "" ]] && [[ "${knownSystemNameInBuildName}" != "" ]] ; then
+  ATDM_SYSTEM_NAME=${knownSystemNameInBuildName}
+  ATDM_HOSTNAME=${systemNameTypeMatchedListHostNames[${ATDM_SYSTEM_NAME}]}
+  assert_selected_system_matches_known_host_in_build_name || return
+  assert_selected_system_matches_known_system_type_mathces || return
+fi
+
+# D.2) Second, go with matches based on hostname
+if [[ "${ATDM_SYSTEM_NAME}" == "" ]] && [[ "${hostnameMatch}" != "" ]] ; then
```

Comment:
Is `hostnameMatch` set by the `assert_selected_system_matches_known_host_in_build_name`? If so, it may make sense (for readability) to use something like `hostnameMatch=assert_selected_system_matches_known_host_in_build_name`.

Reply:
No, the function `assert_selected_system_matches_known_host_in_build_name` reads the var `hostnameMatch` but does not set it.  You can see these functions implemented in the new file `get_system_info_utils.sh`.